### PR TITLE
add method to retrieve last rate limit info per action/method

### PR DESCRIPTION
**Pull request by mydagobah**

### CODE REVIEW

File: sailthru/Sailthru_Client.php [hunk 1]
```diff
@@ -1563,4 +1581,49 @@ protected function prepareJsonPayload(array $data, array $binary_data = array())
...
         return $payload;
     }
+
+    /**
+     * get the rate limit information for the very last call with given action and method
+     * @param string $action
+     * @param string $method GET, POST or DELETE
+     * @return array or null
+     */
+    public function getLastRateLimitInfo($action, $method) {
```
Comment (eweiser) [inline sailthru/Sailthru_Client.php:1591]: @georgekliao wanted us to leave "limit" out of these names, so "getLastRateInfo()", `$rate_info`, etc

File: sailthru/Sailthru_Client.php [hunk 2]
```diff
@@ -1420,37 +1428,47 @@ protected function httpRequestCurl($url, array $data, $method = 'POST', $options
...
         curl_setopt($ch, CURLOPT_HTTPHEADER, $this->httpHeaders);
-        $data = curl_exec($ch);
+        $response = curl_exec($ch);
         $this->lastResponseInfo = curl_getinfo($ch);
+
         curl_close($ch);
-        if (!$data) {
+
+        // parse headers and body
+        $parts = explode("\r\n\r\nHTTP/", $response);
```
Comment (eweiser) [inline sailthru/Sailthru_Client.php:1444]: i assume you got this from [this stackoverflow question](http://stackoverflow.com/a/17971689/396193)? the CURLOPT_HEADERFUNCTION looks kind of cool too, but works for me

File: sailthru/Sailthru_Client.php [hunk 3]
```diff
@@ -50,6 +50,11 @@ class Sailthru_Client {
      */
     private $lastResponseInfo = null;
 
+    /**
+     * Rate Limit information for last API call
+     * @var type
+     */
+    private $lastRateLimitInfo = null;
```
Comment (eweiser) [inline sailthru/Sailthru_Client.php]: initialize this to `[]`?

File: sailthru/Sailthru_Client.php (hunk 2 above)
Comment (mydagobah) [inline sailthru/Sailthru_Client.php:1444]: @eweiser yes, I also tried that solution but didn't work for me. the callback only gets the first line of the headers, then the body decoding still messes up.

File: sailthru/Sailthru_Client.php (hunk 1 above)
Comment (mydagobah) [inline sailthru/Sailthru_Client.php:1591]: feel like it's more clear to use _RateLimit_ to represent the feature. _rate_ itself might be ambiguous. What do you think @georgekliao?

File: sailthru/Sailthru_Client.php (hunk 3 above)
Comment (mydagobah) [inline sailthru/Sailthru_Client.php]: fixed

File: sailthru/Sailthru_Client.php (hunk 1 above)
Comment (georgekliao) [inline sailthru/Sailthru_Client.php:1591]: @eweiser as discussed -- let's have it match the header.

**Comment by eweiser** [comment]
LGTM

**Pull request merged**